db/migrate/20180705230311_smart_proxy_capabilities.rb

```diff
@@ -0,0 +1,23 @@
+class SmartProxyCapabilities < ActiveRecord::Migration[5.1]
+  def up
+    rename_table :features_smart_proxies, :smart_proxy_features
+    change_table :smart_proxy_features do |t|
+      t.primary_key :id
+      t.text :capabilities, array: true
+      if t.respond_to?(:jsonb)
+        t.jsonb :settings
+      elsif t.respond_to?(:json)
+        t.json :settings
+      else
+        t.text :settings
+      end
```

Comment:
Implementation of JSON support in ActiveRecord is poor, we shall avoid this as long as possible. There is no real benefit in storing the data in native JSON, you don't query any data and ActiveRecord is actually slower than just with text implementation.

In other words, unless there is a JSON index created and a JSON query in real use, there is no added benefit in storing JSON natively through ActiveRecord as of today. Note that the SQL JSON syntax is not standardized, that's also a challenge we'd need to solve.

Unless you show good reasons to do that, let's stick with text.

Reply:
Perhaps that was the result of wanting to play with new technology on a Friday. It was still on my list to evaluate this and perhaps fall back to a simple serialized JSON string like on sqlite for reasons you mentioned.